app/src/main/java/com/codingblocks/cbonlineapp/adapters/QuizAttemptListAdapter.kt

```diff
@@ -1,67 +1,57 @@
 package com.codingblocks.cbonlineapp.adapters
 
-import android.content.Context
+import android.graphics.Color
+import android.graphics.Typeface
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.ArrayAdapter
 import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
 import com.codingblocks.cbonlineapp.R
-import com.codingblocks.cbonlineapp.util.OnItemClickListener
 import com.codingblocks.cbonlineapp.extensions.formatDate
 import com.codingblocks.onlineapi.models.QuizAttempt
-import org.jetbrains.anko.AnkoLogger
-import java.util.*
 
-class QuizAttemptListAdapter(internal var context: Context,
-                             var list: ArrayList<QuizAttempt>?,
-                             var listener: OnItemClickListener
-) : ArrayAdapter<QuizAttempt>(context, 0), AnkoLogger {
-
-    override fun getCount(): Int {
-        return if (list == null) 0 else list!!.size
-    }
-
-    override fun getItem(i: Int): QuizAttempt? {
-        return list!![i]
+class QuizAttemptListAdapter(private val dataset: ArrayList<QuizAttempt>) :
+    RecyclerView.Adapter<QuizAttemptListAdapter.QuizAttemptViewHolder>() {
+
+    var onItemClick: ((QuizAttempt)->Unit) ?= null
+
+    inner class QuizAttemptViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
```

Comment:
No need of inner class 
A simple class would be fine

Reply:
done.